internal/adc/translator/apisixroute.go

```diff
@@ -364,14 +389,26 @@ func (t *Translator) translateApisixRouteStreamBackendResolveGranularity(tctx *p
 }
 
 func (t *Translator) translateApisixRouteBackendResolveGranularityEndpoint(tctx *provider.TranslateContext, arNN types.NamespacedName, backend apiv2.ApisixRouteHTTPBackend) (adc.UpstreamNodes, error) {
+	serviceNN := types.NamespacedName{
+		Namespace: arNN.Namespace,
+		Name:      backend.ServiceName,
+	}
+	svc, ok := tctx.Services[serviceNN]
+	if !ok {
+		return nil, errors.Errorf("service not found, ApisixRoute: %s, Service: %s", arNN, serviceNN)
+	}
+	port, err := getPortFromService(svc, backend.ServicePort)
```

Comment:
If I understand correctly, when we resolve ports using endpoints as the granularity, resolving the service port alone is insufficient.

We must further resolve the targetPort used by the pod through the specified service port, and correctly handle cases where the targetPort is either a string or a number?
There is no guarantee that the service port and pod port will use the same value, so directly using the service port is inappropriate.

ping @AlinsRan @ronething